\documentclass[pdftex,english,11pt,parskip=half]{scrartcl}
\usepackage{palatino}
\usepackage{mathpazo}
\usepackage[margin=0.7in]{geometry}
%\usepackage{parskip}
\usepackage[compact]{titlesec}
\usepackage{amsmath,amssymb}
\usepackage{graphicx}
\usepackage{babel}
\usepackage{framed}
\usepackage{wrapfig}
\usepackage{subfig}
\usepackage{sidecap}
\usepackage[labelfont=bf,font=small,format=plain]{caption}
\usepackage{doi}
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{multirow}
\usepackage[table]{xcolor}
\usepackage{wrapfig}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage{array}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\usepackage{float}
%\usepackage[authoryear]{natbib}
\usepackage[numbers]{natbib}
\usepackage{url,hyperref,color}
\usepackage{rotating}
\usepackage{lscape}
\definecolor{darkblue}{rgb}{0.0,0.0,0.75}
\hypersetup{colorlinks,breaklinks,
            linkcolor=darkblue,urlcolor=darkblue,
            anchorcolor=darkblue,citecolor=darkblue}
\newcommand{\fixme}[1]{{\color{red} #1}}
\renewcommand\thesection{\Alph{section}}
\renewcommand{\familydefault}{\sfdefault}
\newcommand{\tabitem}{~~\llap{\textbullet}~~}
\begin{document}
\addtokomafont{section}{\large}
\def\bf{\normalfont\bfseries}
\pagestyle{empty}

\section*{Research Education Program Plan}

\section{Significance}
\vspace{-0.1in}

Current focal points of NIH-funded biomedical research include translational
medicine, big data analyses of genomic, proteomic, and metabolomic experiments,
and team science \cite{}, all of which rely on the generation, sharing, and
integration of high-quality experimental data that may be used and re-used in
multiple types of analysis and visualization platforms. The NIH Strategic Plan
for Data Science \cite{} emphasizes the critical need to capitalize on this
emerging ecosystem of data, and includes the need to address multiple practical
issues related to data storage, management, accessibility, and usablility.
However, consideration of the current sources of much of this data from
individual bench scientists and academic research groups, leads to the 
recognition that a large amount of biomedical data collection is in the form 
of commercial or open source spreadsheets that contain sometimes complex 
organizational
patterns of quantitative and qualitative tables, charts, graphs, notes and other
figures, embedded black-box statistical calculations, and mixtures of
experimental data types and units. Further, it is widely known and discussed
among data scientists, mathematical modelers, and statisticians \cite{}, that
there is a frequent need disgard, transform, and reformat sometimes large
amounts of this primary data and meta-data. While such ``data cleaning'' is
necessary for it's use in the advanced and developing data tools, it is often
conducted by the data-scientist with minimal or no communcation with those who
generated the data. From our experience with drug and vaccine development for
infectious diseases, \textbf{we have identified this step of data recording and
pre-processing as a potential critical point of failure for reproducible
research results.} In this proposal, we provide a solution to mitigate such
potential failures by using the R programming language and the concept of "tidy"
data stuructures, together with a set of training modules that are aimed at the
laboratory bench scientists whose attention is rather to their experimental
technique and collection of accurate data, and who may have little or no
background in the use of general purpose software tools.

While this proposal uses examples from microbiology and immunology experiments,
the collection of primary data in a flexible, open source, transpararent, 
and reproducible format, which can be kept in it's primary state without the 
need for additional modification, is a solution that extends thoughout
large and small scale biomedical science projects.  Figure () below shows an 
example workflow of such a comprehensive experimental data and quantitative 
analysis framework, and indicates our specific focus, highlighting it's 
critical place in such an analysis framework.



\section{Innovation} \vspace{-0.1in}

Our emphasis on primary experimental data collected and recorded as a data
structure, specifically the innovative "tidy" data format developed within the R
statistical programming environment, and presented in an innovative e-book
format



[A paragraph on why we, specifically, passionately think that these training
modules would address a key need.] Our team combines experts in R programming
(Anderson, Lyons) with a group of biomedical researchers (Gonzalez-Juarrero,
Henao-Tamayo, and Robertson) who have, collectively, spent decades in
laboratory-based research to improve understanding of tuberculosis and other
diseases. We met as faculty members of the same College at Colorado State
University, and since have discovered how many of the tools that Drs. Anderson
and Lyons teach and use to improve the reproducibility of \textit{data analysis}
for biomedical research could substantially improve reproducibility and
trasparency in the laboratory-based biomedical research projects of Drs.
Gonzalez-Juarrero, Henao-Tamayo, and Robertson at the stages of \textit{data
recording} and \textit{data pre-processing}. Over the past year, we have begun
to work together to do this within our own research projects: for example, in
Fall 2017 Dr. Gonzalez-Juarrero attended Dr. Anderson's (PI) course in \textit{R
Programming for Research} and has brought the ideas and techniques back to her
research laboratory, in Fall 2017 Dr. Lyons worked with Dr. Anderson to bring in
real tuberculosis drug development data to use in the final group project in Dr.
Anderson's R Programming course, and in Spring 2018 Dr. Henao-Tamayo and Dr.
Anderson began co-advising a graduate student with the aims of implementing
open-source tools for pre-processing flow cytometry in Dr. Henao-Tamayo's
laboratry. Collectively, we are passionate about the idea that
\textbf{open-source tools can be used to bring substantial improvements to
reproducibility of data recording and pre-processing in laboratory-based
research}, and yet we are also able to recognize the key barriers in
implementing these tools in this setting, as well as in training
laboratory-based researchers in how to use these tools, and why existing free
training materials have, to date, been limited in meeting these needs for the
key audience of laboratory-based biomedical researchers. 

[Paragraph on why we are focusing on data recording and data pre-processing.]
Many excellent free training resources exist to improve the computational
reproducibility of biomedical research. However, most of these
materials---including some developed by Dr. Anderson for her open online and
CSU-based courses in R programming [refs]---target researchers at the stage of
\textit{data analysis}, and provide much less guidance on the principles and
techniques to improve reproducibility of the earlier steps of
\textbf{experimental data recording} and \textbf{experimental data
pre-processing}. In this project, we will create training modules to fill this
gap. [More on this, ideally with some references to the importance of
reproduciblity in these steps.]

[A paragraph on who our key audience is and why we are focusing on that
audience.] A key aim is to make these modules \textbf{accessible and useful to
our target audience, laboratory-based researchers}. Three of the Co-Is on our
team (Gonzalez-Juarrero, Henao-Tamayo, and Robertson) represent this key
audience. [Why it is so important to develop these training materials for this
audience.]

[A paragraph about how we plan to focus on this audience] We plan to make these
modules accessible and useful to our target audience, laboratory-based
researchers by including examples from real microbiology and immunology research
projects and by piloting the training modules among laboratory-based biomedical
researchers. Our training materials will provide resources that can be used by
this audience of researchers at a variety of levels, from undergraduate students
to principal investigators, with a modular design that will allow a trainee to
use the subset of modules the best meets his or her immediate needs.

[A paragraph on how our training materials will fit in with the training
materials developed under previous rounds of this grant.]

\section{Innovation} \vspace{-0.1in}

To disseminate the training materials we develop, \textbf{we will use the
innovative \textit{bookdown} framework \cite{xie2016bookdown} to structure and
publish all of our training materials as a free and open online book}. The
\textit{bookdown} framework has been available for a little over two years and
extends the principles of \textit{rmarkdown} to allow users to create attractive
online books that integrate programming code and text. This format allows
authors to efficiency create books with many coding examples---code and its
output does not need to be copied and pasted into a document, but instead can be
automatically generated each time the book is edited. Also, since the book is
created by executing those code examples, regularly test that all code remains
free of bugs created by changes in dependencies (e.g., updates to R packages
used in the code) \cite{xie2016bookdown}. Through this innovative framework, we
will be able to create a searchable online book that weaves R code into the text
and that can include embedded tutorial videos, active weblinks to online
references, and computationally reproducible practice examples and exercises
(Figure \ref{fig:prototype}). While relatively new, this framework has proven
itself reliable and effective---it serves as the framework for several extremely
popular and highly-accessed books on R programming, including \textit{R for Data
Science} [ref]. 

\begin{figure}[t] \includegraphics[width =
\textwidth]{figures/book_prototype.jpg} \caption{Prototype of online course
book, with features highlighted.} \label{fig:prototype} \end{figure}

Dr. Anderson (PI) was an early adopter of the \textit{bookdown} framework. Since
it became available, she has used it to create two \textit{bookdown}-based
books: the \textit{R Programming for Research} online coursebook, which she uses
as a joint textbook and website for the \textit{R Programming for Research}
course she teaches each fall at Colorado State University, and \textit{Mastering
Software Development in R}, which she co-wrote with Dr. Roger Peng as a manual
for developing advanced R programming skills and which has been downloaded by
over [x] people from LeanPub. Dr. Anderson served as one of the six
reviewers---along with R programming experts Jan de Leeuw, Karl Broman, Michael
Grayling, Daniel Kaplan, and Max Kuhn---for \textit{bookdown: Authoring Books
and Technical Documents with R Markdown} \cite{xie2016bookdown}.

\textbf{The use of this framework will help us effectively reach a large
audience in need of training materials for improving the reproducibility of data
recording and pre-processing in biomedical research.} We will be able to freely
post this book online using the Git Pages feature of GitHub, as we have done
with previous books created with the same framework. This will allow anyone to
freely access this content online, and to explore the module contents to find
the set of modules that best suits their immediate training needs. The
book-style format makes the content easy to navigate through both its use of
sections and chapters to group material and through its embedded search
functionality (Figure \ref{fig:prototype}). Further, researchers will be able to
download the book as either a PDF or EPUB file, to use as a reference as they
continue learning to implement reproducibility tools in their research projects.
[More about this.]

We will publish the book's code openly online through GitHub from the beginning
of our development of the training materials, as well as publish the current
version of materials as the online book. \textbf{By making the training
materials available from day one, as they are developed, we will be able to
attract early users to help disseminate the material as it evolves and also be
able to get early feedback on the content as it is developed.} Once the online
book is developed, we will be able to submit a static version of it to be posted
on the homepage of the \textit{bookdown.org} website [ref], which will serve as
another way for trainees to find and access the materials.

\section{Proposed Research Education Program}

\textbf{Project goals and objectives.} Our project's primary goal is to develop
training modules that address the needs of laboratory-based biomedical
researchers seeking to improve reproducibility, especially of experimental data
recording and pre-processing, in their research projects. The expected result of
this project is an online book that contains [x] training modules as separate
chapters, with video lectures, written text, and additional educational
materials collected within each module's chapter. We consider it critical that
these training materials be clear, relevant, and useful to a key audience of
biomedical scientists whose primary research activities focus on laboratory
research, rather than data analysis or statistics. To help us achieve this goal,
our team includes three co-investigators from this key audience who will play an
early and continuing role in developing and refining the training materials for
this audience, and our plan includes intensive user testing of the developing
materials on several groups of pilot testers from this audience. Our
co-investigators' network of connections in Microbiology and Immunology will
help us widely disseminate these materials to our target audience, as will the
free, online format of the materials and presenting the materials at national
and international Microbiology and Immunology conferences in the second and
third years of the project.

\subsubsection*{Proposed training modules}

\begin{quotation} ``State the \textbf{goals for education} and \textbf{justify
the area of training} selected for module development in terms of its
\textbf{relevance and potential impact} on improving the development of skills
and knowledge important for conducting rigorous and reproducible research.
Describe the \textbf{subject material} to be covered. The \textbf{length} of the
proposed training module should be explained in terms of \textbf{scope and depth
of coverage} of the subject matter.  In addition, \textbf{how the research
education will be utilized by trainees or investigators} should be
described---for example, a module on how to avoid confirmation bias to be taken
by all beginning laboratory workers, or a module on appropriate design of animal
studies to be taken immediately prior to beginning such work." \end{quotation}

We aim to create training modules that will show laboratory-based biomedical
researchers how simple computational reproducibility principles can improve
biomedical research reproducibility at the stages of data recording and
preprocessing and teach them how to implement these principles in their
laboratories. The importance of computational reproducibility of scientific
research is increasingly recognized by scientists, journals, and funding
agencies, with such ``computationally reproducible" research requiring that all
data and code for a research project be available and that this data and code
can be used to regenerate study findings either by the original researcher or by
other researchers \cite{ellis2017share, ram2013git}. Among our team, we have
found that there are many common existing practices---including use of
spreadsheets with embedded macros to concurrently record and analyze
experimental data, unstructured [too strong?] management of project files,
reliance on proprietory, vendor-supplied point-and-click software for data
pre-processing---that can interfere with the transparency, reproducibility, and
efficiency of the kind of research they conduct. [Some more about how we're not
the first people to notice that this is a problem for reproducibility / rigor.] 

We propose to develop two collections of modules, \textbf{Improving the
Reproducibility of Experimental Data Recording} and \textbf{Improving the
Reproducibility of Experimental Data Pre-Processing}. Our team has worked
together to create a curriculum of training modules that we believe will help
fill an important training gap for laboratory-based biomedical researchers
(Tables \ref{tab:content_one} and \ref{tab:content_two}). The first sequence,
\textbf{``Improving the Reproducibility of Experimental Data Recording"; Table
\ref{tab:content_one}}, will explore the pitfalls of combining experimental data
recording and analysis within macro-enabled spreadsheets, explain the power
structured data formats for recording data, describe how reproducibility can be
improved by using a single structured directory to store all research project
files, and demonstrate the use of version control to maintain single, current
versions of all files while saving a history of all file changes. The second
sequence, \textbf{``Improving the Reproducibility of Experimental Data
Pre-Processing"; Table \ref{tab:content_two}}, will focus on improving the
reproducibility of experimental data pre-processing steps, like gating for flow
cytometry data and peak finding / quantifying for mass spectrometry data.
Training materials will explain how the use of code scripts for these steps
dramatically improves reproducibility compared to using vendor-supplied
point-and-click software and will introduce trainees to popular R software for
this pre-processing. This sequence will include advice on reproducible data
pre-processing protocols and how to create them using literate programming tools
(\textit{Rmarkdown}).

Each module will fall into one of three categories for teaching reproducibility:
(1) principals; (2) implementation; and (3) case study examples. ``Principals"
modules will be programming-language agnostic, while ``Implementation" modules
will focus on tools available through the popular open source R software and its
RStudio interface. Working with the biomedical laboratory-based co-investigators
on our team, we will ensure that these modules and the examples used in them are
approachable and useful to researchers with limited computational training. We
have divided the content into modules in a way that will allow \textbf{trainees
and investigators at any level} to create their own ``tracks" by selecting
relevant subsets of the modules to complete, and potentially combining this
content with other training modules available through the NIH's [Clearinghouse]
[ref]. Table \ref{tab:tracks} gives a few examples of how different trainees
could create and follow their own ``track" of the material.

For many of the barriers we identified to reproducibility at these stages of
research, a simple fix exists. Our team has worked together to craft a list of
specific topics (Tables \ref{tab:content_one} and \ref{tab:content_two}) we aim
to address with these modules so that they will offer training in a collection
of principles and techniques that tackle barriers to reprodubility with
straightforward fixes, but which we have found are still very common in
biomedical laboratory-based research programs, including those here at Colorado
State University. We have identified several easy-to-address barriers to
computational reproducibility at two stages of biomedical research: experimental
data recording and experimental data pre-processing.  

\underline{\textbf{Improving the Reproducibility of Experimental Data
Recording}} 

The first sequence will provide principles and tools for increasing the
computational reproducibility at an early stage of biomedical research, as
experimental data is recorded. This sequence will include [x] modules covering
four main topics: 

\textbf{Separating data recording and analysis.} (Figure \ref{fig:spreadsheet})
... This topic will covered in a \textit{Principles} module on ``Separating data
recording and analysis".

\begin{figure}[b] \centering \includegraphics[width =
0.8\textwidth]{figures/algorithms.png} \caption{An \textit{xkcd} cartoon
captures the ballooning complexity and lack of transparency that can result from
many people using a spreadsheet with embedded macros. This practice can be a
critical barrier to reproducibility and transparency in biomedical research, yet
is still a common practice in many biomedical laboratories. \textit{Source: xkcd
by Randall Munroe}} \label{fig:spreadsheet} \end{figure}

\textbf{Using a structured data format to record data.} Every extra step of data
formatting is another chance to introduce an error in the data. Therefore, by
keeping research data pipelines simple---which can be more easily achieved if
data is initially recorded in a format amenable to later data pre-processing,
analysis, and visualization---researchers can decrease the potential for errors
in the data and therefore improve the rigor and reproducibility of their
research. One key concept for improving the reproducibility of experimental data
collection is understanding how to create and use structured data formats, of
which the ``tidy" data format is one popular implementation. The ``tidy" data
format plugs into the \textit{tidyverse} framework, which enables powerful and
user-friendly data management, processing, and analysis by combining simple
tools to solve complex, multi-step problems, and this framework is enabled by
ensuring those simple tools share a common interface: a ``tidy" data format
\cite{ross2017declutter, silge2016tidytext, wickham2016ggplot2, wickham2016r}.
Working within the R framework facilitates research that adheres to standards of
reproducibility through scriptable data analysis that can easily be placed under
version control \cite{bryan2017excuse}. Since the tools are simple and share a
common interface, they are easier to learn, use, and combine than tools created
in the classical R framework \cite{ross2017declutter, lowndes2017our,
reviewer2017review, mcnamara2016state}. This \textit{tidyverse} framework is
quickly becoming the standard taught in introductory R courses and books
\cite{hicks2017guide, baumer2015data, kaplan2017teaching, stander2017enthusing,
reviewer2017review, mcnamara2016state}, ensuring ample training resources for
researchers new to programming, including books (e.g., \cite{baumer2017modern,
lifesciencesR}, some freely available online, e.g., \cite{wickham2016r}),
massive open online courses (MOOCs), onsite university courses
\cite{baumer2015data, kaplan2017teaching, stander2017enthusing}, and Software
Carpentry workshops \cite{wilson2014software, pawlik2017developing}. Further,
tools that extend the tidyverse have been created to enable high-quality data
analysis and visualization in several domains, including text mining
\cite{silge2017text}, microbiome studies \cite{mcmurdie2013phyloseq}, natural
language processing \cite{RJ-2017-035}, network analysis \cite{RJ-2017-023},
ecology \cite{hsieh2016inext}, and genomics \cite{yin2012ggbio}. This topic will
be covered in a \textit{Principles} module on ``Principles and power of
structured data formats", two \textit{Implementation} modules on ``The 'tidy'
data format: an implementation of a structured data format" and ``Designing
templates for tidy data collection", and one \textit{Example} module called
``Example: Creating a template for 'tidy' data collection".

\textbf{Managing all research project files in a single, structured directory.}
Reproducbility can also be improved, starting at the data recording stage or
earlier, by using single, structured directory to store all files related to the
project. This ``project" framework has recently been encouraged by a number of
researchers as a way to enable computationally reproducible research, especially
for research conducted by teams \cite{marwick2017packaging,
parker2017opinionated, lowndes2017our}. RStudio allows users to create their own
custom ``Project" template, suited to a specific type of data analysis or
software development, which can then be registered and accessed by other users
\cite{rstudioprojecttemplate}. While a ``Project" can have any internal
structure, a common structure can be enforced for a certain type of project
through the creation of a new ``Project" template, which defines the required
subdirectories, structure, and file names of common elements that must exist in
the project \cite{rstudioprojecttemplate}. This template, when selected by a
future user, will create a new directory with a ``skeleton" structure,
potentially including templated files (e.g., for metadata). This topic will be
covered in a \textit{Principles} module on the ``Power of using a single
structured 'Project' directory for storing and tracking research project files",
an \textit{Implementation} module on ``Creating 'Project' templates", and an
\textit{Example} module called ``Example: Creating a 'Project' template".

\textbf{Implementing version control.} The use of \textit{Git} and
\textit{GitHub} has also been encouraged as a tool to enable reproducible
research \cite{piccolo2016tools, ram2013git, bryan2017excuse, lowndes2017our,
cetinkaya2017infrastructure}. Projects saved in a single, structured directory
can be easily put under \textit{Git} version control in \textit{RStudio}, which
includes a pane that allows users to work under version control without learning
command-line version control language and, if desired, easily connect the
project with an online version of the project hosted on \textit{GitHub}.  This
topic will be covered in two \textit{Principles} modules called ``Harnessing
version control for transparent data recording" and ``Enhance the
reproducibility of collaborative research with version control platforms" as
well as an \textit{Implementation} module on ``Using git and GitLab to implement
version control".

\underline{\textbf{Improving the Reproducibility of Experimental Data
Pre-Processing}}

The second sequence will provide principles and tools for increasing the
computational reproducibility at an early stage of biomedical research, as
experimental data is pre-processed. This sequence will include [x] modules
covering three main topics: 

\textbf{Using code scripts to pre-process experimental data.} Scriptable
software tools bring key advantages compared to GUI software in terms of data
pre-processing \cite{cetinkaya2017infrastructure, huber2015orchestrating,
preeyanon2014reproducible, piccolo2016tools}, but it is critical to provide some
training on the use of these tools for researchers new to programming. Expertise
with a scripting language is not universal across the biomedical community,
although literacy in programming is increasing in the sciences
\cite{ram2013git}, and many now recommend programming as a critical skill for
all biology Ph.D. students \cite{list2017ten}. This topic will be covered in one
\textit{Principles} module on ``Principles and benefits of scripted
pre-processing of experimental data" and two \textit{Implementation} modules
called ``Introduction to scripted data pre-processing in R" and ``Simplify
scripted pre-processing through R's 'tidyverse' tools".

\textbf{Working with complex data types during pre-processing.} ... This topic
will be covered in a \textit{Principles} module on ``Complex data types in
experimental data pre-processing", a \textit{Implementation} module on ``Complex
data types in R and Bioconductor", and an \textit{Example} module called
``Example: Converting from complex to 'tidy' data formats".

\textbf{Reproducible data pre-processing protocols.} ... This topic will be
covered in a \textit{Principles} module called ``Introduction to reproducible
data pre-processing protocols", an \textit{Implementation} module on ``RMarkdown
for creating reproducible data pre-processing protocols", and an
\textit{Example} module called ``Example: Creating a reproducible data
pre-processing protocol". 

\underline{\textbf{Choice of tools to include in ``Implementation" modules.}}

To improve reproducibility practices among our target audience, it is important
that we provide them with some instruction on how to implement the
reproduciblity principles we cover. For some of these principles, there are
several reasonable and well-developed tools that could be used for
implementations. However, few researchers are interested in learning every tool,
and instead would prefer to learn one set of tools that ``just work". 

[We chose R because it is a top-notch programming language for biomedical
research applications.] We have chosen for the implementation portion of these
modules to focus on tools from the open-source R programming language. R can be
freely, quickly, and easily downloaded and installed to a user's computer,
allowing new users to get started quickly, a critical consideration for usable
scientific software \cite{list2017ten}. R has been maintained for over a decade
by the R Development Core Team and works with all major computing platforms,
ensuring  widespread access, stability, and compatability, also critical for
ease-of-use \cite{baumer2017lessons, altschul2013anatomy}. R offers a
well-developed environment for creating new tools that extend the core language
\cite{wickham2015r} and includes ample tools for documenting research workflows
\cite{xie2015dynamic, xie2016bookdown}. R's status as the \textit{lingua franca}
of statisticians and biostatisticians means that its use in early stages of
experimental data recording and pre-processing can help foster closer
collaborations between laboratory-based scientists and statisticians throughout
the research process. R can be scaled as the volume of data in projects grows
\cite{list2017ten}, as it includes tools to interface with distributed computing
platforms (e.g., \textit{Hadoop} \cite{pathak2014rhadoop}, \textit{Spark}
\cite{sparklyr}), and its scripts can be integrated within workflow management
systems (e.g., \textit{Galaxy} \cite{goecks2010galaxy, walker2016models}). Some
of the implementation tools---git and GitLab, for example---are separate from R
but can be mastered much more easily if trainees are taught to use them through
RStudio's user-friendly interfaces rather than using the command line or other
alternative interfaces.

[We chose R because we know how to use it and effectively teach it to a wide
range of researchers] We have also selected to focus on this set of tools for
the \textit{Implementation} modules because two members of our team (Anderson
and Lyons) use R daily for their own research and so are able to provide much
better instruction and details on using this set of tools than alternative tools
for the same tasks. Dr. Anderson also regularly teaches students in her
department how to use this set of tools through a graduate-level course and has
developed techniques for helping students new to programming to master R using
the RStudio interface, including using RStudio's interface to use RMarkdown,
track a project under git version control, and connect to version control
platforms to improve collaborative work. 

[Much of our material can be used by a researcher without R programming
experience.] We appreciate that many researchers do not know the R language, and
some of them may want to learn more about improving reproducibility without
needing to learn a new programming language. For this reason, we have
deliberately separated the ``Principles" and ``Examples" content in our modules
from the ``Implementation" modules, so that a researcher can select a track of
our modules that does not require programming knowledge. Further, while all of
the ``Implementation" modules are conceptually focused on tools that an R
programmer would use, several of the modules could be appreciated and used to
improve reproducibility without a mastery of R. For example, RStudio and its
``Projects" functionality can be used to help manage research project files,
keep them under version control, and interface with GitLab to work
collaboratively without using any R code within the project. Similarly, while
the ``tidy" data format is currently an important implementation common within R
for structuring data, understanding its principals and characteristics does not
require any knowledge of R. In our table of example ``tracks" that a trainee
could follow, the track for an example Principal Investigator gives one example
of a track that would not require any prior knowledge of R or other programming
languages.   

\input{tables/module_content_data_recording.tex}

\input{tables/module_content_data_preprocessing.tex}

\input{tables/module_tracks.tex}

\subsubsection*{Format for the training modules}

\begin{quotation} ``Describe the \textbf{format} for the training module
proposed and \textbf{justify it in terms of the education goals}."
\end{quotation}

\textbf{Online book.} We will use an online book to collect all the training
materials in a single structure. Each chapter of the book will contain all the
materials from one of the modules listed in Tables \ref{tab:content_one} and
\ref{tab:content_two}, for [x] chapters total. We have created a prototype
(Figure \ref{fig:prototype}) to demonstrate some features of the final book.
Users will be able to quickly navigate through chapters with a navigation bar on
the left of the webpage, with chapter subsection links opening when a chapter is
selected. We will embed the lecture video at the start of the chapter, allowing
a user to watch the video content without leaving the book website, which
prevent having to ``hop around" online to watch the video and then access
written text and additional educational materials in the book. The book will
include a link for the trainee to download a copy as a PDF or EPUB file, to use
as a future reference offline if desired. The format also includes buttons that
can be used to share the link to the online book with others through Twitter and
other platforms, as well as a link to the book's GitHub repository, to allow
early users of the in-development materials to provide feedback on typos, broken
links, unclear materials, and other issues to iron out as we develop the
materials. 

We will create this online book using the \textit{bookdown} framework
\cite{xie2016bookdown}. This in an innovative framework that will allow us to
create a searchable online book that weaves R code into the text and that can
include embedded tutorial videos, active weblinks to online references, and
computationally reproducible practice examples and exercises. Further, by
including R code examples as executable code, we will be able to use this online
book to frequently check tutorial code examples to quickly identify and fix any
broken tutorial code \cite{xie2016bookdown}. We will use GitHub's Git Pages to
freely post this book online. Dr. Anderson (PI) has previously created two
\textit{bookdown}-based books, \textit{R Programming for Research} and
\textit{Mastering Software Development in R}, and has posted and maintained
\textit{R Programming for Research} online continuously since Fall 2016 through
Git Pages.  

\textbf{Video lectures.} Each chapter will include a video lecture that covers
the module's material, with the approximate length of each lecture listed in
Tables \ref{tab:content_one} and \ref{tab:content_two}. We will record these
video lectures in CSU's Computer Assisted Teaching Support laboratory (see
letter from Dr. West), which includes equipment and staff for creating
professional-quality video lectures. We will use YouTube to freely host these
videos and embed the video within the text of that module's chapter in the
online book (see Figure \ref{fig:prototype} for an example of how this will look
to trainees). Trainees will be able to watch each of these videos without
leaving the online books webpage, while hosting them through YouTube will allow
us to take advantage of their excellent, free, and well-tested platform for
sharing videos, as well as allow us to collect detailed analytics on how often
each video is watched and for how long, to help us assess the use of this
component of the training material. 

\textbf{Additional educational materials.} Each module will contain additional
educational material, to help the trainee absorb the material and assess his or
her mastery of the topics. Depending on the module, this additional content will
either be a quiz, questions for discussion, or an applied exercise to try out
the implementation of a tool or principle covered in the module (see Tables
\ref{tab:content_one} and \ref{tab:content_two} for the specific material
planned for each module). To help engage trainees, we will include audio and
video content walking the trainee through answers and solutions for these
additional materials. While not a substitute for in-person training, these video
and audio discussions of the materials is meant to mimic the detailed
walk-throughs and discussions that we would do after a student attempted these
materials if we were teaching these materials in person. We will tape the video
and audio content in CSU's Computer Assisted Teaching Support laboratory (see
letter from Dr. West). We will host the video content through YouTube and the
audio content through SoundCloud and embed this content in the online book, as
with the video lectures. We will use Google Forms as a free and unlimited way
for us to create the quizzes and embed them in the online book [ref].

\noindent \textbf{Insuring compliance with Rehabilitation Act.} We have plans
for making our proposed training module section 508 compliant of the
Rehabilitation Act (29 U.S.C. '794 d), as amended by the Workforce Investment
Act of 1998. Much of the online content will be text based. For figures and
other images in the book, we will use alt and \textit{longdesc} attributes
within the image tag to provide a text alternative. Using YouTube to host the
lecture videos will allow us to draw on their functionality for accessibility,
including ``enough time" requirements in terms of being able to pause and turn
off the content. YouTube allows users to add and edit closed caption content on
their videos, which we will use to add optional closed captions to this
content---in addition to improving accessibility of the content, it may also
help trainees for whom English is a second language to follow the content. In
the first two years of the grant, we will have a student hourly who will assist
Dr. Anderson in the technical implementation of the online book, and helping to
ensure the content is compliant with the Rehabilitation Act will be one of her
key tasks. If this task requires the use of interesting techniques or
technologies, Dr. Anderson and the student may prepare a journal article
describing these techniques and submit it to \textit{The R Journal} during the
project period. The R community is very open to and interested in improving
accessibility, as evidenced by previous publications, presentations, and
software on these topics (e.g., [ref, uswebr ref]).

\subsection{Team}

\input{tables/team_description.tex}

Our team (Table \ref{tab:team_description}) combines experts in R programming (Anderson, Lyons), including its use
to improve the computational reproducibility of health-related research, with
laboratory-based academic researchers in Microbiology and Immunology
(Henao-Tamayo, Gonzalez-Juarrero, Robertson) who are \textbf{attuned to the
needs of and barriers to improving the reproducibility of experimental data
collection and pre-processing among laboratory-based biomedical researchers}.
Our team will allow us to develop training modules that present state-of-the-art
approaches and tools for reproducibility, but do so in a way that is prioritized
to be most useful and accessible to health researchers whose training has
focused on laboratory-related, rather than computational, methods, and for whom
existing training materials on computational reproducibility might be hard to
understand or apply to their own research projects. Our team also includes a
CSU-based expert in program evaluation (Maertens), to assist us in planning and
implementing evaluations of the training materials.

\noindent Our team will also include \textbf{Dr. Julie Maertens (Research Associate)}, a senior evaluator
at the Colorado State University STEM Center, which assists and collaborates
with faculty involved with STEM education-based research and programming in
designing and carrying out evaluations. She will assist in the design and
implementation of project evaluation throughout this project. Dr. Maertens will
only be involved in the project to assist in planning and implementing
evaluation, and her percent effort is capped at 0.85\% to reflect the RFA's
budget restriction of \$3,000 total on program evaluation, including salary
support.

\subsubsection*{Coordination and management of the team}

\input{tables/team_roles.tex}

Our plan of content development and publishing is ambitious, but we are
confident we can meet it. Dr. Anderson previously developed the online content,
in collaboration with a co-instructor, for a five-course specialization with
approximately twice as much content in under nine months. She has experience
developing and publishing online training materials for learning R-based tools,
including through the \textit{bookdown} online book framework we plan to use
here. In this previous development of online training materials, Dr. Anderson
learned to collaborate closely with co-authors in developing and refining the
content and in developing quizzes and applied exercises to allow trainees to
test their understanding of the content. She will bring that experience in the
proposed project, to help organize and integrate the contributions of our team
of co-investigators. For the technical development of the book, she will be
assisted by and supervise an undergraduate hourly during the first two years of
the project, to help in turning the intellectual content that she and the
co-investigators develop into the formatted online book. She will work with Dr.
Maertens to solidify and implement evaluations of the training materials.

\subsection{Institutional Environment and Commitment}

Colorado State University is very supportive of improving the reproducbility of
research, and it offers ample resources that will help us ensure the success of
the proposed project. \textbf{Our institution has clearly expressed its support
for our proposed effort to create, refine, evaluate, and disseminate these
training materials} in institutional letters of support from Dr. Jac Nickoloff
(Chair of the Department of Environmental \& Radiological Health Sciences, CSU),
Dr. Dean Gregg (Chair of the Department of Microbiology, Immunology, \&
Pathology, CSU), and Dr. Alan Rudolph (Vice President for Research, CSU). As
expressed in the letter of support from Dr. Nickoloff, Dr. Anderson's teaching
expectations during the project period are capped at three courses every two
years, allowing adequate time for her to complete the tasks required by this
proposal. As also expressed in that letter, Dr. Anderson's department supports
that the training materials developed under this grant will be made freely
available through online publication under a Creative Commons license. Dr.
Anderson's department has supported her previous development and publication of
similarly free and open training materials following a similar format [ref].

Colorado State University is host to a large number of laboratory-based
biomedical researchers, especially in fields related to drug and vaccine
development for infectious diseases. We will \textbf{take advantage of this
environment} to help us pilot test and refine the training materials, to ensure
they are clear, relevant, and useful to our target population of
laboratory-based biomedical researchers. We have received statements of support
from relevant programs, including the the director of the CSU Interdisciplinary
Graduate Program in Cell \& Molecular Biology, and Associate Director of the
Interdisciplinary NSF-NRT GAUSSI training program in Computational Biology, to
help disseminate our materials to pilot testers and early users of our training
materials within CSU (see letters from Drs. Dean Gregg, Carol Wilusz, and Jeff
Wilusz). 

Colorado State University offers many excellent facilities that we will take
advantage of to help us achieve our project's goals. CSU provides meeting rooms
that we can reserve free-of-charge to use for the CSU-based user testings. The
Computer Assisted Teaching Support (CATS) laboratory at CSU's Academy for
Teaching and Learning has professional-grade recording equipment that we will
use to record the video lectures within each module (see letter from Dr. Andrew
West). CSU's STEM Center will provide help in evaluating the training modules,
including through the budgeted involvement of Dr. Julie Maertens, a senior
evaluator at the center. 

\subsection{Piloting and evaluating the effectiveness of the training modules}

We will conduct two, day-long user testing sessions at CSU in each year of the
project. The CSU user testing groups will consist of current and future
laboratory-based biomedical researchers. We will recruit trainees with a variety
of research roles, including undergraduate students, graduate students,
postdoctoral fellows, research associates, and principal investigators. We have
informed several CSU biomedical researchers about these proposed testing
sessions and received their support in encouraging CSU researchers within their
groups and departments to participate (see letters from ...). 

Table \ref{tab:evaluation}.

In the first two project years, each session will test the set of modules
developed since the last user testing (approximately [x] modules will be tested
each of these session). The sessions will begin with our team giving live
lectures of the same content we plan to film for the video lectures included in
the online book. This will allow us to improve and refine this content, based on
detailed feedback from testers representative of our target audience, before
filming the final video lectures. During the rest of the session, we will divide
the trainees into small teams to work through the additional educational
materials (applied exercises, quiz questions, and discussion questions) to iron
out problems with the clarity or implementation of these materials. The trainees
will have access to the in-development online book as they work through these
materials.

[Information we will collect from these training sessions.]

Dr. Anderson (PI) has experience in productively conducting these kinds of user
testing sessions at CSU. She has run several two-hour user testing sessions with
students from various departments of Colorado State University prior to
releasing R software packages \cite{futureheatwaves, countyweather}. Further, in
April 2016, she led a longer, two-day user testing session through a Weather
Data Hackathon at Colorado State University (Figure \ref{csu-r-hackathon}).
Around 15 people participated, including undergraduate students, graduate
students, postdoctoral fellows, and professors from CSU's Departments of
Atmospheric Sciences, Civil \& Environmental Engineering, Microbiology, and
Statistics. Some of the ideas and code developed during this Hackathon have
since led to development and publication of open source software
\cite{countyfloods, noaastormevents}.

\begin{SCfigure} \centering \includegraphics[width =
0.6\textwidth]{figures/csu_hackathon.png} \caption{Some of the approximately 15
undergraduate students, graduate students, postdoctoral fellows, and professors
who participated in a two-day Weather Data Hackathon at Colorado State
University in April 2016 led by Dr. Anderson.} \label{csu-r-hackathon}
\end{SCfigure}

\begin{quotation} ``Applications must include a plan for evaluating the
activities supported by the award in terms of their \textbf{frequency of use}
and their \textbf{usefulness}. The use of \textbf{multiple evaluation
approaches} is highly encouraged as is \textbf{testing several groups with
different characteristics}. The application must specify \textbf{baseline
metrics (e.g., numbers, educational levels, and demographic characteristics of
test group)} in a structured format, as well as \textbf{measures to gauge the
short and long-term success of the research education award in achieving its
objectives}. Applicants are expected to \textbf{obtain feedback from test group}
to help identify weaknesses and to provide suggestions for improvements, and
\textbf{make the evaluation and feedback data} available to NIGMS staff."
\end{quotation}

\input{tables/evaluation.tex}

\textbf{Learning objectives} These are what we're trying to determine were achieved by the training modules.

\textbf{Pilot / text group evaluation}:

\textbf{Long-term evaluation}:

\begin{itemize}
\item Google Analytics for online book. How often are people accessing the book? How long are they spending on the book website? Where are the people accessing the book?
\item YouTube analytics for the embedded videos. How often are people accessing the book? How long are they spending on the book website? Where are the people accessing the book?
\item Quiz for each chapter of the book? Use to evaluate how well they've mastered the material? (Possibly could use embedded Shiny apps for this? Other ways to do this?)
\item Rating options for each chapter of the online book? Usefulness? What they learned?
\item Survey within each chapter of the online book? Educational level, demographic characteristics.
\end{itemize}

\textbf{Things we want to know about the final training materials:}

\textit{Quantitative:}

\begin{itemize}
\item How many people have accessed the online book?
\item How are book users distributed across the U.S.? Are there many international users accessing the book?
\item When someone accesses the online book, how long on average do they spend reading or exploring the book? How many users are spending more than 10 minutes on the book per visit (about the minimum time for a module)?
\item How many people have downloaded the entire book?
\item How many people have commented on or made suggestions for the book through its ``Issues" page?
\item How many people have viewed each of the tutorial videos?
\item What percent of people who begin to watch each video finish it? 
\item For each video, are there common locations in the video where many people turn it off?
\item For online quizzes, how many people take each quiz?
\item For each quiz question, what percent of people answer it correctly? Are there some questions that many trainees struggle with after going through the training material?
\item For audio files of Co-Is answering discussion questions, how often do people listen to those? How many people listen to the complete file?
\item For applied exercise that include data or files to download, how many people have downloaded the files?
\item What is the demographic profile of users (age, gender, race, ...)?
\item What is the educational profile of users (highest degree, area of research)?
\item What is the professional profile of users (current position, typical research tasks for which the trainee is responsible)?
\item For each module, which elements did a trainee use (online book text, video tutorial, additional content like discussion questions and quizzes)? How useful did the trainee find each of the elements he or she used (on a Likert scale)?
\item For each module, did the user feel that he or she had adequate prior knowledge to follow the material in the module, or were there unstated prerequisites that the trainee lacked and felt limited their ability to learn from the training module?
\item For each module, does the trainee anticipate making any changes to his / her research practice based on having taken this module?
\item For each module, what percent of the material was new to the trainee (versus things he or she had already learned or heard about outside of the module)?
\end{itemize}

\textit{Qualitative:}

\begin{itemize}
\item What types of suggestions and comments have people made about the book through its ``Issues" page? Are these mostly to fix typos, or are there substantive questions? Have users helped to identify areas with which they struggled?
\item What are the users' goals in taking the training modules?
\item How did the user decide which modules to take?
\item For each module, what are elements that the trainee wished had been covered but were not?
\item For each module, were there elements that the trainee did not find useful that could have been excluded?
\item For each module, how did the trainee use any additional materials (discussion questions, quiz questions, applied exercises) that came with the module?
\item For each module, how does the trainee plan to change his or her research practice based on having taken the module?
\item Did the trainee consider using other training materials to learn this material (e.g., on-campus courses, online MOOCs, other books or video series)? If so, what aspects of our training materials led to the decision to pick them?
\item Why did the trainee decide to use these training modules (e.g., requirement for a course, PI told them to, self-motivated to improve their research practices, interested in learning new technology)?
\end{itemize}

\textbf{Levels of evaluation:}

\begin{itemize}
\item \textbf{Final users of the online book and videos.} These could potentially be from anywhere in the world, and for many we won't have great ways to contact them. 
\item \textbf{Workshop attendees for the workshops we plan to propose and do at national microbiology / immunology conferences.} For these people, we could definitely do a survey before to get information on demographics, education level, interest in the training materials, etc. We could also do a post survey to find out what they learned, how helpful it was, what they found confusing, etc. Finally, we could get their email addresses to ask some longer-term evaluation questions (e.g., How are they using what they learned 1--2 years after taking the workshop? How much did they retain in terms of principals, implementation, and examples?). We can also use questions that are asked during the workshops and areas where additional materials (applied exercises, quizzes, discussion questions) are problematic to help us hone our training materials.
\item \textbf{Early online users.} We will plan to develop and post the text and some of the additional educational materials (e.g., quizzes, discussion questions) online through GitHub \textit{as we write the book and develop the materials}. We will use social media to invite people to try out the book as we develop it. Based on previous work developing online books, we have found that this open development process can help attract users very early in the process, and that these users are often very helpful in providing feedback as the book is developed. We will elicit their feedback through GitHub (``Issues" page will be the main forum for them to post comments and suggestions).
\item \textbf{CSU pilot users.} We can ask these pilot users many questions, both before and after the pilot testing. Further, we will have access to ask them longer-term outcomes, as well as to ask at the department level how the use of these training materials by a number of people in the department has changed research practices and what is considered ``best practice" for research in the department (i.e., a `bubble up" effect).
\item \textbf{Pilot users from other institutions.} Similar to CSU pilot users, although we'll have a bit less access for determining longer-term and department-wide outcomes.
\end{itemize}
    

\subsection{Dissemination Plan}

\begin{quotation}
``A specific plan must be provided to disseminate the finished training modules \textbf{nationally} and make them \textbf{freely accessible}. In addition, links to these modules will be posted and maintained on the NIGMS web site."
\end{quotation}

Actions we will take to disseminate these training materials include: 

\begin{itemize}
\item Publish online, openly and freely, as an online book with a Creative Commons license. All video and audio content will be published online in free formats through YouTube and SoundCloud, with the content embedded in the online books webpage. 
\item It will be easy to link to this online book through a link posted on the NIGMS web site [Clearinghouse page].
\item Post a static version of the book to bookdown.org, where people can go to find free online books published using the bookdown format. 
\item Advertise through social media, and have our networks of collaborators (in biology and in R) help us get out the word through this medium.
\item Apply to present a workshop in year 2 of the project at [microbiology conference] to pilot the training materials and help get out the news among our target audience that these materials are freely available 
\item Apply to present posters or oral presentations in years 2 and 3 of the project at three national and international conferences ([conference names]) to pilot the training materials and help get out the news among our target audience that these materials are freely available
\item Pilot materials among CSU researchers in our target audience through biannual, day-long user testing. In addition to providing us with feedback to help refine our materials, this will help us let members of our target audience know that these training materials are freely available and how to find them.
\item Invite colleagues (and their research group members) at non-CSU institutions to serve as early pilot testers of the online version of all our materials. In addition to providing us with feedback to help refine our materials, this will help us let members of our target audience know that these training materials are freely available and how to find them.
\end{itemize}

We will create an online tutorial book, since providing tutorials, example code, and example datasets can substantially improve the ability of new users to learn software tools \cite{list2017ten}. We will use GitHub's free ``Pages" website hosting to publish the book freely online, and we will also submit it to the \textit{bookdown.org} website under the Creative Commons Attribution-ShareAlike 4.0 International License [ref]. Dr. Anderson (PI) has previously created two \textit{bookdown}-based books, \textit{R Programming for Research} and \textit{Mastering Software Development in R}. Both are publicly and freely available online under the Creative Commons license (see LOS, Xie). 

We will publish the video lectures using the YouTube platform and embed these videos within the online book. The videos, like the book, will be published under a Creative Commons license.

For many online training materials on the principals and implementation of computationally reproducible research, the target audience is statisticians and other researchers who focus on data analysis. This audience now has access to many excellent resources for improving research reproducibility at the data analysis stage. Our target audience is instead researchers who focus on conducting experimental, laboratory-based biomedical research and whose research tasks are more focused on the earlier research steps of running experiments, recording experimental data, and pre-processing that data, prior to data analysis steps. We are focusing our dissemination plans on this target audience, for whom training materials on improving the reproducibility of later data analysis steps might be of limited relevance.

We will also take steps to make sure that our target audience---laboratory-based biomedical scientists---are aware of these training materials. We will travel in Years 02 and 03 to one national microbiology ([conference]) and one national immunology ([conference]) conference. We will submit proposals to these conferences to present half-day (student?) workshops covering why and how to improve reproducibility in experimental data recording and pre-processing for biomedical microbiology and immunology research. We will also submit poster abstracts to present and discuss the training materials as part of each conference's poster sessions. We have budgeted for two members of our team (the PI and one co-I) to attend each of these conferences to help disseminate the training materials produced by the project. [CSU's microbiology seminar series? Conferences on-campus at CSU? Are there ones that are repeated every year?]

The PI has previously had substantial success in disseminating online training materials. She is the co-instructor of a five-course specialization on \textit{Mastering Software Development in R} through the Massive Open Online Course platform Coursera. The first course in this series has had over 25,000 participants since it was opened in fall 2016. An accompanying online book on the LeanPub platform has been accessed by [x] people. The PI, however, does not have a presence in the microbiology and immunology community, and so the co-investigators, who are heavily involved in this community, will form a key bridge in making sure our target audience is aware that these training materials are available.

\subsection{Timeline}

\begin{figure}[ht]
    \includegraphics[width=\textwidth]{figures/timeline.pdf}
    \caption{Timeline for proposed activities for this project.}
    \label{fig:timeline}
\end{figure}

\clearpage

\bibliographystyle{unsrtnat}
\bibliography{rep_modules}

\end{document}
