\documentclass[pdftex,english,11pt,parskip=half]{scrartcl}
\usepackage{palatino}
\usepackage{mathpazo}
\usepackage[margin=0.7in]{geometry}
%\usepackage{parskip}
\usepackage[compact]{titlesec}
\usepackage{amsmath,amssymb}
\usepackage{graphicx}
\usepackage{babel}
\usepackage{framed}
\usepackage{wrapfig}
\usepackage{subfig}
\usepackage{sidecap}
\usepackage[labelfont=bf,font=small,format=plain]{caption}
\usepackage{doi}
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{multirow}
\usepackage[table]{xcolor}
\usepackage{wrapfig}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage{array}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\usepackage{float}
%\usepackage[authoryear]{natbib}
\usepackage[numbers]{natbib}
\usepackage{url,hyperref,color}
\usepackage{rotating}
\usepackage{lscape}
\definecolor{darkblue}{rgb}{0.0,0.0,0.75}
\hypersetup{colorlinks,breaklinks,
            linkcolor=darkblue,urlcolor=darkblue,
            anchorcolor=darkblue,citecolor=darkblue}
\newcommand{\fixme}[1]{{\color{red} #1}}
\renewcommand\thesection{\Alph{section}}
\renewcommand{\familydefault}{\sfdefault}
\newcommand{\tabitem}{~~\llap{\textbullet}~~}
\begin{document}
\addtokomafont{section}{\large}
\def\bf{\normalfont\bfseries}
\pagestyle{empty}

\section*{Research Education Program Plan}

\section{Significance}
\vspace{-0.1in}

[A paragraph on why we, specifically, passionately think that these training modules would address a key need.] Our team combines experts in R programming (Anderson, Lyons) with a group of biomedical researchers (Gonzalez-Juarrero, Henao-Tamayo, and Robertson) who have, collectively, spent decades in laboratory-based research to improve understanding of tuberculosis and other diseases. We met as faculty members of the same College at Colorado State University, and since have discovered how many of the tools that Drs. Anderson and Lyons teach and use to improve the reproducibility of \textit{data analysis} for biomedical research could substantially improve reproducibility and trasparency in the laboratory-based biomedical research projects of Drs. Gonzalez-Juarrero, Henao-Tamayo, and Robertson at the stages of \textit{data recording} and \textit{data pre-processing}. Over the past year, we have begun to work together to do this within our own research projects: for example, in Fall 2017 Dr. Gonzalez-Juarrero attended Dr. Anderson's (PI) course in \textit{R Programming for Research} and has brought the ideas and techniques back to her research laboratory, in Fall 2017 Dr. Lyons worked with Dr. Anderson to bring in real tuberculosis drug development data to use in the final group project in Dr. Anderson's R Programming course, and in Spring 2018 Dr. Henao-Tamayo and Dr. Anderson began co-advising a graduate student with the aims of implementing open-source tools for pre-processing flow cytometry in Dr. Henao-Tamayo's laboratry. Collectively, we are passionate about the idea that \textbf{open-source tools can be used to bring substantial improvements to reproducibility of data recording and pre-processing in laboratory-based research}, and yet we are also able to recognize the key barriers in implementing these tools in this setting, as well as in training laboratory-based researchers in how to use these tools, and why existing free training materials have, to date, been limited in meeting these needs for the key audience of laboratory-based biomedical researchers. 

[Paragraph on why we are focusing on data recording and data pre-processing.] Many excellent free training resources exist to improve the computational reproducibility of biomedical research. However, most of these materials---including some developed by Dr. Anderson for her open online and CSU-based courses in R programming [refs]---target researchers at the stage of \textit{data analysis}, and provide much less guidance on the principles and techniques to improve reproducibility of the earlier steps of \textbf{experimental data recording} and \textbf{experimental data pre-processing}. In this project, we will create training modules to fill this gap. [More on this, ideally with some references to the importance of reproduciblity in these steps.]

[A paragraph on who our key audience is and why we are focusing on that audience.] A key aim is to make these modules \textbf{accessible and useful to our target audience, laboratory-based researchers}. Three of the Co-Is on our team (Gonzalez-Juarrero, Henao-Tamayo, and Robertson) represent this key audience. [Why it is so important to develop these training materials for this audience.]

[A paragraph about how we plan to focus on this audience] We plan to make these modules accessible and useful to our target audience, laboratory-based researchers by including examples from real microbiology and immunology research projects and by piloting the training modules among laboratory-based biomedical researchers. Our training materials will provide resources that can be used by this audience of researchers at a variety of levels, from undergraduate students to principal investigators, with a modular design that will allow a trainee to use the subset of modules the best meets his or her immediate needs.

[A paragraph on how our training materials will fit in with the training materials developed under previous rounds of this grant.]

\section{Innovation}
\vspace{-0.1in}

To disseminate the training materials we develop, \textbf{we will use the innovative \textit{bookdown} framework \cite{xie2016bookdown} to structure and publish all of our training materials as a free and open online book}. The \textit{bookdown} framework has been available for a little over two years and extends the principles of \textit{rmarkdown} to allow users to create attractive online books that integrate programming code and text. This format allows authors to efficiency create books with many coding examples---code and its output does not need to be copied and pasted into a document, but instead can be automatically generated each time the book is edited. Also, since the book is created by executing those code examples, regularly test that all code remains free of bugs created by changes in dependencies (e.g., updates to R packages used in the code) \cite{xie2016bookdown}. Through this innovative framework, we will be able to create a searchable online book that weaves R code into the text and that can include embedded tutorial videos, active weblinks to online references, and computationally reproducible practice examples and exercises (Figure \ref{fig:prototype}). While relatively new, this framework has proven itself reliable and effective---it serves as the framework for several extremely popular and highly-accessed books on R programming, including \textit{R for Data Science} [ref]. 

\begin{figure}[t]
\includegraphics[width = \textwidth]{figures/book_prototype.jpg}
\caption{Prototype of online course book, with features highlighted.}
\label{fig:prototype}
\end{figure}

Dr. Anderson (PI) was an early adopter of the \textit{bookdown} framework. Since it became available, she has used it to create two \textit{bookdown}-based books: the \textit{R Programming for Research} online coursebook, which she uses as a joint textbook and website for the \textit{R Programming for Research} course she teaches each fall at Colorado State University, and \textit{Mastering Software Development in R}, which she co-wrote with Dr. Roger Peng as a manual for developing advanced R programming skills and which has been downloaded by over [x] people from LeanPub. Dr. Anderson served as one of the six reviewers---along with R programming experts Jan de Leeuw, Karl Broman, Michael Grayling, Daniel Kaplan, and Max Kuhn---for \textit{bookdown: Authoring Books and Technical Documents with R Markdown} \cite{xie2016bookdown}.

\textbf{The use of this framework will help us effectively reach a large audience in need of training materials for improving the reproducibility of data recording and pre-processing in biomedical research.} We will be able to freely post this book online using the Git Pages feature of GitHub, as we have done with previous books created with the same framework. This will allow anyone to freely access this content online, and to explore the module contents to find the set of modules that best suits their immediate training needs. The book-style format makes the content easy to navigate through both its use of sections and chapters to group material and through its embedded search functionality (Figure \ref{fig:prototype}). Further, researchers will be able to download the book as either a PDF or EPUB file, to use as a reference as they continue learning to implement reproducibility tools in their research projects. [More about this.]

We will publish the book's code openly online through GitHub from the beginning of our development of the training materials, as well as publish the current version of materials as the online book. \textbf{By making the training materials available from day one, as they are developed, we will be able to attract early users to help disseminate the material as it evolves and also be able to get early feedback on the content as it is developed.} Once the online book is developed, we will be able to submit a static version of it to be posted on the homepage of the \textit{bookdown.org} website [ref], which will serve as another way for trainees to find and access the materials.

\section{Approach}

\begin{quotation}
``Does the proposed program clearly state its goals and objectives, including the audience to be reached, the content to be conveyed, and the intended outcome?  Is there evidence that the program is based on a sound rationale, as well as sound educational concepts and principles? Is the plan for evaluation sound and likely to provide information on the effectiveness of the program?"
\end{quotation}

\subsection{Proposed Research Education Program Plan}

\begin{quotation}
``While the proposed research education program may complement ongoing research training and education occurring at the applicant institution, the proposed educational experiences must be distinct from those research training and research education programs currently receiving federal support. When research training programs are on-going in the same department, the applicant organization should clearly distinguish between the activities in the proposed research education program and the research training supported by the training program. The research education proposed must be \textbf{targeted to trainees and investigators at any level}. State the \textbf{goals for education} and \textbf{justify the area of training} selected for module development in terms of its \textbf{relevance and potential impact} on improving the development of skills and knowledge important for conducting rigorous and reproducible research. Describe the \textbf{subject material} to be covered.  Describe the \textbf{format} for the training module proposed and \textbf{justify it in terms of the education goals}.  The \textbf{length} of the proposed training module should be explained in terms of \textbf{scope and depth of coverage} of the subject matter.  In addition, \textbf{how the research education will be utilized by trainees or investigators} should be described---for example, a module on how to avoid confirmation bias to be taken by all beginning laboratory workers, or a module on appropriate design of animal studies to be taken immediately prior to beginning such work.  Describe the \textbf{plans for piloting and evaluating the effectiveness} of the training module. Describe \textbf{plans for making the proposed training module section 508 compliant of the Rehabilitation Act} (29 U.S.C. '794 d), as amended by the Workforce Investment Act of 1998 (P.L. 105 – 220; see http://www.section508.gov/ for additional information). Provide a \textbf{timeline for module development, piloting and refinement, dissemination, evaluation, and maintenance}.  This timeline must propose \textbf{making the training publicly available within two years} of the award date."
\end{quotation}

\subsubsection{Educational goals of the modules}

The importance of computational reproducibility of scientific research is increasingly recognized by scientists, journals, and funding agencies, with such ``computationally reproducible" research requiring that all data and code for a research project be available and that this data and code can be used to regenerate study findings either by the original researcher or by other researchers \cite{ellis2017share, ram2013git}.

\textbf{Improving the Reproducibility of Experimental Data Recording} Every extra step of data formatting is another chance to introduce an error in the data. Therefore, by keeping research data pipelines simple---which can be more easily achieved if data is initially recorded in a format amenable to later data pre-processing, analysis, and visualization---researchers can decrease the potential for errors in the data and therefore improve the rigor and reproducibility of their research.

One key concept for improving the reproducibility of experimental data collection is understanding how to create and use structured data formats, of which the ``tidy" data format is one popular implementation. The ``tidy" data format plugs into the \textit{tidyverse} framework, which enables powerful and user-friendly data management, processing, and analysis by combining simple tools to solve complex, multi-step problems, and this framework is enabled by ensuring those simple tools share a common interface: a ``tidy" data format \cite{ross2017declutter, silge2016tidytext, wickham2016ggplot2, wickham2016r}. Working within the R framework facilitates research that adheres to standards of reproducibility through scriptable data analysis that can easily be placed under version control \cite{bryan2017excuse}. Since the tools are simple and share a common interface, they are easier to learn, use, and combine than tools created in the classical R framework \cite{ross2017declutter, lowndes2017our, reviewer2017review, mcnamara2016state}. This \textit{tidyverse} framework is quickly becoming the standard taught in introductory R courses and books \cite{hicks2017guide, baumer2015data, kaplan2017teaching, stander2017enthusing, reviewer2017review, mcnamara2016state} (see also Letters of Support [LOS], Kimmel, Peng), ensuring ample training resources for researchers new to programming, including books (e.g., \cite{baumer2017modern, lifesciencesR}, some freely available online, e.g., \cite{wickham2016r}), massive open online courses (MOOCs), onsite university courses \cite{baumer2015data, kaplan2017teaching, stander2017enthusing}, and Software Carpentry workshops \cite{wilson2014software, pawlik2017developing}. Further, tools that extend the tidyverse have been created to enable high-quality data analysis and visualization in several domains, including text mining \cite{silge2017text}, microbiome studies \cite{mcmurdie2013phyloseq}, natural language processing \cite{RJ-2017-035}, network analysis \cite{RJ-2017-023}, ecology \cite{hsieh2016inext}, and genomics \cite{yin2012ggbio}.

Reproducbility can also be improved, starting at the data recording stage or earlier, by using single, structured directory to store all files related to the project. RStudio allows users to create their own custom ``Project" template, suited to a specific type of data analysis or software development, which can then be registered and accessed by other users \cite{rstudioprojecttemplate}. While a ``Project" can have any internal structure, a common structure can be enforced for a certain type of project through the creation of a new ``Project" template, which defines the required subdirectories, structure, and file names of common elements that must exist in the project \cite{rstudioprojecttemplate}. This template, when selected by a future user, will create a new directory with a ``skeleton" structure, potentially including templated files (e.g., for metadata). Projects saved in this format can be easily put under \textit{Git} version control in \textit{RStudio}, which includes a pane that allows users to work under version control without learning command-line version control language and, if desired, easily connect the project with an online version of the project hosted on \textit{GitHub}. This ``project" framework has recently been encouraged by a number of researchers as a way to enable computationally reproducible research, especially for research conducted by teams \cite{marwick2017packaging, parker2017opinionated, lowndes2017our}, and the use of \textit{Git} and \textit{GitHub} has also been encouraged as a tool to enable reproducible research \cite{piccolo2016tools, ram2013git, bryan2017excuse, lowndes2017our, cetinkaya2017infrastructure}.  

\textbf{Improving the Reproducibility of Experimental Data Pre-Processing.} Scriptable software tools bring key advantages compared to GUI software in terms of data pre-processing \cite{cetinkaya2017infrastructure, huber2015orchestrating, preeyanon2014reproducible, piccolo2016tools}, but it is critical to provide some training on the use of these tools for researchers new to programming. Expertise with a scripting language is not universal across the biomedical community, although literacy in programming is increasing in the sciences \cite{ram2013git}, and many now recommend programming as a critical skill for all biology Ph.D. students \cite{list2017ten}. 

\textbf{Teaching implementation tools to improve reproducibility.} These modules will teach the principals of reproducibility as well as introduce researchers to tools for implementing reproducible research workflows. The implementation portion of these modules will focus on tools from the open-source R programming language. R can be freely, quickly, and easily downloaded and installed to a user's computer, allowing new users to get started quickly, a critical consideration for usable scientific software \cite{list2017ten}. R has been maintained for over a decade by the R Development Core Team and works with all major computing platforms, ensuring  widespread access, stability, and compatability, also critical for ease-of-use \cite{baumer2017lessons, altschul2013anatomy}. R offers a well-developed environment for creating new tools that extend the core language \cite{wickham2015r} and includes ample tools for documenting research workflows \cite{xie2015dynamic, xie2016bookdown}. R's status as the \textit{lingua franca} of statisticians and biostatisticians means that its use in early stages of experimental data recording and pre-processing can help foster closer collaborations between laboratory-based scientists and statisticians throughout the research process. R can be scaled as the volume of data in projects grows \cite{list2017ten}, as it includes tools to interface with distributed computing platforms (e.g., \textit{Hadoop} \cite{pathak2014rhadoop}, \textit{Spark} \cite{sparklyr}), and its scripts can be integrated within workflow management systems (e.g., \textit{Galaxy} \cite{goecks2010galaxy, walker2016models}). 

\subsubsection{Module subject material}

We propose to develop two collections of modules, \textbf{Improving the Reproducibility of Experimental Data Recording} and \textbf{Improving the Reproducibility of Experimental Data Pre-Processing}. Our team has worked together to create a curriculum of training modules that we believe will help fill an important training gap for laboratory-based biomedical researchers (Tables [x] and [x]). The first sequence, \textbf{``Improving the Reproducibility of Experimental Data Recording"; Table [x]}, will explore the pitfalls of combining experimental data recording and analysis within macro-enabled spreadsheets, explain the power structured data formats for recording data, describe how reproducibility can be improved by using a single structured directory to store all research project files, and demonstrate the use of version control to maintain single, current versions of all files while saving a history of all file changes. The second sequence, \textbf{``Improving the Reproducibility of Experimental Data Pre-Processing"; Table [x]}, will focus on improving the reproducibility of experimental data pre-processing steps, like gating for flow cytometry data and peak finding / quantifying for mass spectrometry data. Training materials will explain how the use of code scripts for these steps dramatically improves reproducibility compared to using vendor-supplied point-and-click software and will introduce trainees to popular R software for this pre-processing. This sequence will include advice on reproducible data pre-processing protocols and how to create them using literate programming tools (\textit{Rmarkdown}).

Each module will fall into one of three categories for teaching reproducibility: (1) principals; (2) implementation; and (3) case study examples. ``Principals" modules will be programming-language agnostic, while ``Implementation" modules will focus on tools available through the popular open source R software and its RStudio interface. Working with the biomedical laboratory-based co-investigators on our team, we will ensure that these modules and the examples used in them are approachable and useful to researchers with limited computational training. 

We have divided the content into modules in a way that will allow different trainees to create their own ``tracks" by selecting a relevant subset of the full set of modules, and potentially combining this content with other training modules available through the NIH's [Clearinghouse] [ref]. Table [x] gives a few examples of how different trainees could create and follow their own ``track" of the material. 

\input{tables/module_content_data_recording.tex}

\input{tables/module_content_data_preprocessing.tex}

\input{tables/module_tracks.tex}

\subsubsection{Format for the training modules}

\begin{itemize}
\item Online book created through the ``bookdown" format, with each module as a book chapter. We can use Git Pages to host this (CSU options for web hosting?).
\item Training videos embedded for each module, each 5--30 minutes. Videos will be similar to online course lectures and will be hosted using YouTube. Embedding in the book will allow users to watch videos without leaving the book's webpage. 
\item Each chapter will end with exercise questions (around 10 questions, combination of discussion questions and applied exercises), as well as an embedded video with discussion of the discussion questions and a detailed walk-through of answers to applied exercises. 
\item Possibly host this through an online course platform like DataCamp?
\end{itemize}

\textbf{Online book.} To ensure that these training modules are easy for researchers to access, use, and reference, we will provide all training materials through an online book created with the \textit{bookdown} framework \cite{xie2016bookdown} (see LOS, Xie). Through this new framework, we will be able to create a searchable online book that weaves R code into the text and that can include embedded tutorial videos, active weblinks to online references, and computationally reproducible practice examples and exercises. Further, by including R code examples as executable code, we will be able to use this online book to frequently check tutorial code examples to quickly identify and fix any broken tutorial code \cite{xie2016bookdown}.  Dr. Anderson (PI) has previously created two \textit{bookdown}-based books, \textit{R Programming for Research} and \textit{Mastering Software Development in R}.  

\subsubsection{Piloting and evaluating effectiveness of training modules}

We will conduct two, day-long user testing sessions at CSU in each year of the project. The CSU user testing groups will consist of current and future laboratory-based biomedical researchers. We will recruit trainees with a variety of research roles, including undergraduate students, graduate students, postdoctoral fellows, research associates, and principal investigators. We have informed several CSU biomedical researchers about these proposed testing sessions and received their support in encouraging CSU researchers within their groups and departments to participate (see letters from ...). 

In the first two project years, each session will test the set of modules developed since the last user testing (approximately [x] modules will be tested each of these session). The sessions will begin with our team giving live lectures of the same content we plan to film for the video lectures included in the online book. This will allow us to improve and refine this content, based on detailed feedback from testers representative of our target audience, before filming the final video lectures. During the rest of the session, we will divide the trainees into small teams to work through the additional educational materials (applied exercises, quiz questions, and discussion questions) to iron out problems with the clarity or implementation of these materials. The trainees will have access to the in-development online book as they work through these materials.

[Information we will collect from these training sessions.]

Dr. Anderson (PI) has experience in productively conducting these kinds of user testing sessions at CSU. She has run several two-hour user testing sessions with students from various departments of Colorado State University prior to releasing R software packages \cite{futureheatwaves, countyweather}. Further, in April 2016, she led a longer, two-day user testing session through a Weather Data Hackathon at Colorado State University (Figure \ref{csu-r-hackathon}). Around 15 people participated, including undergraduate students, graduate students, postdoctoral fellows, and professors from CSU's Departments of Atmospheric Sciences, Civil \& Environmental Engineering, Microbiology, and Statistics. Some of the ideas and code developed during this Hackathon have since led to development and publication of open source software \cite{countyfloods, noaastormevents}.

\begin{SCfigure}
\centering
\includegraphics[width = 0.6\textwidth]{figures/csu_hackathon.png}
\caption{Some of the approximately 15 undergraduate students, graduate students, postdoctoral fellows, and professors who participated in a two-day Weather Data Hackathon at Colorado State University in April 2016 led by Dr. Anderson.}
\label{csu-r-hackathon} 
\end{SCfigure}

\subsubsection{Insuring compliance with Rehabilitation Act}

``Electronic content shall conform to Level A and Level AA Success Criteria and Conformance Requirements in Web Content Accessibility Guidelines (WCAG) 2.0"

\begin{itemize}
\item The vast majority of our content outside of videos will be text based. Even example data files will mostly be text files.
\item Is R itself compliant?
\item ``All non-text content that is presented to the user has a text alternative that serves the equivalent purpose, except for the situations listed below.": `` If a short description can serve the same purpose and present the same information as the non-text content: Providing short text alternative for non-text content that serves the same purpose and presents the same information as the non-text content using one of the following techniques" ``Using alt attributes on img elements. When using the img element, specify a short text alternative with the alt attribute." (We can use this for any figures included in the online book.)
\item ``If a short description can not serve the same purpose and present the same information as the non-text content (e.g., a chart or diagram): Providing short text alternatives that provide a brief description of the non-text content using one of the following techniques" ``Using longdesc".
\item ``If non-text content is a control or accepts user input:" (is this true for the buttons on the book page? Anywhere else we would have this? Quizzes? Surveys?) ``Using alt attributes on images used as submit buttons" (does bookdown already do this?)
\item All quizzes, survey questions, and discussion questions will be text-based. All applied exercises will have text-based instructions.
\item For audio files: include transcript?
\item For videos: include transcript / closed captioning?
\item All audio and video content will be pre-recorded (nothing live)
\item ``Prerecorded Audio-only: An alternative for time-based media is provided that presents equivalent information for prerecorded audio-only content."
\item ``Prerecorded Video-only: Either an alternative for time-based media or an audio track is provided that presents equivalent information for prerecorded video-only content."
\item ``Captions are provided for all prerecorded audio content in synchronized media, except when the media is a media alternative for text and is clearly labeled as such." ``Providing closed captions"
\item ``An alternative for time-based media or audio description of the prerecorded video content is provided for synchronized media, except when the media is a media alternative for text and is clearly labeled as such." ``Providing a second, user-selectable, audio track that includes audio descriptions"
\item ``Enough time" requirements---all should be handled through the platforms we use to host audio and video content (e.g., YouTube). Allows the user to turn off, pause, etc. the content.
\item ``Web pages do not contain anything that flashes more than three times in any one second period, or the flash is below the general flash and red flash thresholds."
\item ``Navigational mechanisms that are repeated on multiple Web pages within a set of Web pages occur in the same relative order each time they are repeated, unless a change is initiated by the user."
\item ``Input Assistance"---may be through hosts of survey / quiz questions?
\end{itemize}

\textit{uswebr:} ``An R package that includes an RMarkdown template (currently) targeted for scientific reports and documentation using the bookdown package's html\_document2 function. The look and feel is based on the US Digital Web Standards. An important component of the US Web Design standards is accessibility. All of the designs meet the WCAG 2.0 AA accessibility guidelines and are compliant with Section 508 of the Rehabilitation Act. Plots and other elements added by the user should continue to follow accessiblity guidelines."

\subsection{Team}

Our team combines experts in R programming (Anderson, Lyons), including its use to improve the computational reproducibility of health-related research, with laboratory-based academic researchers in Microbiology and Immunology (Henao-Tamayo, Gonzalez-Juarrero, Robertson) who are \textbf{attuned to the needs of and barriers to improving the reproducibility of experimental data collection and pre-processing among laboratory-based biomedical researchers}. Our team will allow us to develop training modules that present state-of-the-art approaches and tools for reproducibility, but do so in a way that is prioritized to be most useful and accessible to health researchers whose training has focused on laboratory-related, rather than computational, methods, and for whom existing training materials on computational reproducibility might be hard to understand or apply to their own research projects. Our team also includes a CSU-based expert in program evaluation (Maertens), to assist us in planning and implementing evaluations of the training materials.

\noindent \textbf{Dr. Brooke Anderson (PI)} is an Assistant Professor of Epidemiology in the Department of Environmental \& Radiological Health Sciences at Colorado State University, with an affiliate position at the Department of Statistics. She is an expert in R programming and has created and published several open-source R packages, in particular to facilitate environmental epidemiological research. She has experience creating R programs to work with large data, including climate model output and large weather datasets, as well as programs that interface with open web-based datasets. She is the co-instructor of a series of Massive Open Online Courses on \textit{Mastering Software Development in R} through Coursera and an associated open online book. Dr. Anderson will lead the development and refinement of all training modules developed through this grant, including through supervising the development and integration of training materials from co-investigators. She will also lead user testing and other evaluation of all developed modules to ensure the developed modules are clear, effective, and well-matched to meet the needs of biological researchers from a variety of scientific backgrounds, including those new to programming. She will coordinate the contributions of all other Co-Is in helping to evaluate and disseminate the training materials. She will travel to conferences in Years 02 and 03 of the project to help disseminate the materials to our key audience. Our budget includes funding for a student hourly in the first two years of the project, who Dr. Anderson will supervise as he or she assists in the technical development and online publishing of the online book that will contain all training materials.

\noindent \textbf{Dr. Michael Lyons (Co-I)} is an Assistant Professor at Colorado State University (CSU), where he works on the computational biology and pharmacology of tuberculosis (TB) infection and treatment in experimental animal models and TB patients. Prior to joining CSU full-time in 2011, he was a software engineer in the computer industry for 12 years, and prior to that, a theoretical physicist. Through a K25 award, he obtained significant classroom and hands-on training and exposure to laboratory methods related to drug and vaccine development for TB, providing him with a solid understanding of how preclinical and clinical data are used for evidence-based decision making in the biomedical sciences. He is highly attuned to the problems that this project aims to address, and he has a clear understanding of the practical limitations and challenges for both the laboratory scientist and data analyst. In this project, he will be one of the co-authors of the online book containing the training modules. He will contribute to the development, testing, and refinement of materials for all training modules, with a particularly strong role in helping to create the ``Implementation" and ``Principles" modules. He will help plan and attend all of the biannual day-long pilot testing sessions at CSU. He will help disseminate the final training materials among our target audience. 

\noindent \textbf{Dr. Marcela Henao-Tamayo (Co-I)} is an Assistant Professor at Department of Microbiology, Immunology \& Pathology, College of Veterinary Medicine and Biomedical Sciences, and the Co-Director of CSU-Flow Cytometry Facility at Colorado State University. She studies the immunopathogenesis of tuberculosis using animal models to evaluate the role of different types of T cells and Myeloid Derived cells in tuberculosis and BCG vaccination (Bacille Calmette Guerin, the only approved vaccine against TB). She has tested numerous vaccine candidates evaluating the immune response they elicit in association with protection against tuberculosis disease. She is interested in how existing tools for computational reproducibility can be applied to data recording and pre-processing in her own research laboratory, and she and Dr. Anderson (PI) co-advise a graduate student who is integrating open-source R software into the regular practice of Dr. Henao-Tamayo's research work, including through implementation of reproducible automated gating of flow cytometry data. In this project, she will be one of the co-authors of the online book containing the training modules. She will contribute to the development, testing, and refinement of materials for all training modules, with a particularly strong role in helping to create the ``Principles" and ``Examples" modules, with some of the ``Example" modules focused on improving reproducibility in her own research projects, particularly in the sequence on data pre-processing. She will serve as one of the first testers of the ``Implementation" modules, to help us determine their clarity, relevance, and usefulness for a laboratory-based scientist. She will attend the biannual day-long pilot testing sessions at CSU, and members of her research group will attend these sessions as pilot testers. She will help disseminate the final training materials among our target audience, including through helping prepare the workshops and presentations to be given at conferences during the grant to help disseminate the materials. 

\noindent \textbf{Dr. Mercedes Gonzalez-Juarrero (Co-I)} is an Associate Professor in the Department of Microbiology, Immunology \& Pathology at Colorado State University. Her research interest is in studying the basic nature of the cell mediated immune response to mycobacteria infections. During the last ten years, her research group has undertaken studies to investigate the emergence of immunosuppression during pulmonary tuberculosis, with the primary goal of learning how and where to target the latently infected host to fully recover the antimicrobial activity of the infected cell, and how to use this information in the context of current chemotherapeutic and multidrug resistant TB infections. Dr. Gonzalez-Juarrero became particularly interested in how to improve the reproducibility, transparency, and efficiency of experimental data recording within her research projects when she attended Dr. Anderson's (PI) CSU course on \textit{R Programming for Research} in Fall 2017 and learned about the principles of structured data formats, including the ``tidy" data format now popular with statisticians. This realization led to important discussions with Drs. Anderson and Lyons regarding data recording and understanding the role of each scientist from data recording through to analysis. In this project, she will be one of the co-authors of the online book containing the training modules. She will contribute to the development, testing, and refinement of materials for all training modules, with a particularly strong role in helping to create the ``Principles" and ``Examples" modules, with some of the ``Example" modules focused on improving reproducibility in her own research projects, particularly in Sequence 1 (data recording). She will serve as one of the first testers of the ``Implementation" modules, to help us determine their clarity, relevance, and usefulness for a laboratory-based scientist. She will attend the biannual day-long pilot testing sessions at CSU, and members of her research group will attend these sessions as pilot testers. She will help disseminate the final training materials among our target audience, including through helping prepare the workshops and presentations to be given at conferences during the grant to help disseminate the materials.

\noindent \textbf{Dr. Gregory Robertson (Co-I)} is an Assistant Professor of Microbiology, Immunology and Pathology at Colorado State University. Dr. Robertson has more than 20 years of classical and clinical microbiology experience with emphasis in antibacterial discovery and mode-of-action studies for novel and existing classes of antimicrobials. This includes efforts in academia, and also with larger pharmaceutical corporations (Eli Lilly and Co) and smaller bio-pharmaceutical groups (Cumbre Pharmaceuticals). His current research is focused on \textit{Mycobacterium tuberculosis} host-pathogen interactions and the development and application of novel preclinical animal models to further anti-tuberculosis drug development and evaluate drug resistance. In the context of improving reproducibility in biomedical research, Dr. Robertson is particularly passionate about the perils of using spreadsheets with embedded macros as a tool for recording and analyzing experimental data. In this project, he will be one of the co-authors of the online book containing the training modules. He will contribute to the development, testing, and refinement of materials for all training modules, with a particularly strong role in helping to create the ``Principles" and ``Examples" modules, with some of the ``Example" modules focused on improving reproducibility in her own research projects. He will serve as one of the first testers of the ``Implementation" modules, to help us determine their clarity, relevance, and usefulness for a laboratory-based scientist. He will attend the biannual day-long pilot testing sessions at CSU, and members of his research group will attend these sessions as pilot testers. He will help disseminate the final training materials among our target audience, including through helping prepare the workshops and presentations to be given at conferences during the grant to help disseminate the materials. 

\noindent \textbf{Dr. Julie Maertens (Research Associate)} is a senior evaluator at the Colorado State University STEM Center, which assists and collaborates with faculty involved with STEM education-based research and programming in designing and carrying out evaluations. She will assist in the design and implementation of project evaluation throughout this project. Dr. Maertens will only be involved in the project to assist in planning and implementing evaluation, and her percent effort is capped at 0.85\% to reflect the RFA's budget restriction of \$3,000 total on program evaluation, including salary support.

\subsubsection{Coordination and management of the team}

\begin{quotation}
``Describe \textbf{arrangements for administration} of the program.  Provide evidence that the Program Director/Principal Investigator is actively engaged in research and/or teaching in an area related to the mission of NIH, and can \textbf{organize, administer, monitor, and evaluate the research education program}. For programs proposing multiple PDs/PIs, describe the complementary and integrated expertise of the PDs/PIs; their leadership approach, and governance appropriate for the planned project."
\end{quotation}

\subsection{Institutional Environment and Commitment}

\begin{quotation}
``Describe the institutional environment, reiterating the \textbf{availability of facilities and educational resources} (described separately under Facilities \& Other Resources), that can contribute to the planned Research Education Program. Evidence of institutional commitment to the research educational program is required. A \textbf{letter of institutional commitment} must be attached as part of Letters of Support (see below). Appropriate institutional commitment should include the provision of adequate staff, facilities, and educational resources that can contribute to the planned research education program."
\end{quotation}

\subsection{Evaluation Plan}

\begin{quotation}
``Applications must include a plan for evaluating the activities supported by the award in terms of their \textbf{frequency of use} and their \textbf{usefulness}. The use of \textbf{multiple evaluation approaches} is highly encouraged as is \textbf{testing several groups with different characteristics}. The application must specify \textbf{baseline metrics (e.g., numbers, educational levels, and demographic characteristics of test group)} in a structured format, as well as \textbf{measures to gauge the short and long-term success of the research education award in achieving its objectives}. Applicants are expected to \textbf{obtain feedback from test group} to help identify weaknesses and to provide suggestions for improvements, and \textbf{make the evaluation and feedback data} available to NIGMS staff."
\end{quotation}

\input{tables/evaluation.tex}

\textbf{Learning objectives} These are what we're trying to determine were achieved by the training modules.

\textbf{Pilot / text group evaluation}:

\textbf{Long-term evaluation}:

\begin{itemize}
\item Google Analytics for online book. How often are people accessing the book? How long are they spending on the book website? Where are the people accessing the book?
\item YouTube analytics for the embedded videos. How often are people accessing the book? How long are they spending on the book website? Where are the people accessing the book?
\item Quiz for each chapter of the book? Use to evaluate how well they've mastered the material? (Possibly could use embedded Shiny apps for this? Other ways to do this?)
\item Rating options for each chapter of the online book? Usefulness? What they learned?
\item Survey within each chapter of the online book? Educational level, demographic characteristics.
\end{itemize}

\textbf{Things we want to know about the final training materials:}

\textit{Quantitative:}

\begin{itemize}
\item How many people have accessed the online book?
\item How are book users distributed across the U.S.? Are there many international users accessing the book?
\item When someone accesses the online book, how long on average do they spend reading or exploring the book? How many users are spending more than 10 minutes on the book per visit (about the minimum time for a module)?
\item How many people have downloaded the entire book?
\item How many people have commented on or made suggestions for the book through its ``Issues" page?
\item How many people have viewed each of the tutorial videos?
\item What percent of people who begin to watch each video finish it? 
\item For each video, are there common locations in the video where many people turn it off?
\item For online quizzes, how many people take each quiz?
\item For each quiz question, what percent of people answer it correctly? Are there some questions that many trainees struggle with after going through the training material?
\item For audio files of Co-Is answering discussion questions, how often do people listen to those? How many people listen to the complete file?
\item For applied exercise that include data or files to download, how many people have downloaded the files?
\item What is the demographic profile of users (age, gender, race, ...)?
\item What is the educational profile of users (highest degree, area of research)?
\item What is the professional profile of users (current position, typical research tasks for which the trainee is responsible)?
\item For each module, which elements did a trainee use (online book text, video tutorial, additional content like discussion questions and quizzes)? How useful did the trainee find each of the elements he or she used (on a Likert scale)?
\item For each module, did the user feel that he or she had adequate prior knowledge to follow the material in the module, or were there unstated prerequisites that the trainee lacked and felt limited their ability to learn from the training module?
\item For each module, does the trainee anticipate making any changes to his / her research practice based on having taken this module?
\item For each module, what percent of the material was new to the trainee (versus things he or she had already learned or heard about outside of the module)?
\end{itemize}

\textit{Qualitative:}

\begin{itemize}
\item What types of suggestions and comments have people made about the book through its ``Issues" page? Are these mostly to fix typos, or are there substantive questions? Have users helped to identify areas with which they struggled?
\item What are the users' goals in taking the training modules?
\item How did the user decide which modules to take?
\item For each module, what are elements that the trainee wished had been covered but were not?
\item For each module, were there elements that the trainee did not find useful that could have been excluded?
\item For each module, how did the trainee use any additional materials (discussion questions, quiz questions, applied exercises) that came with the module?
\item For each module, how does the trainee plan to change his or her research practice based on having taken the module?
\item Did the trainee consider using other training materials to learn this material (e.g., on-campus courses, online MOOCs, other books or video series)? If so, what aspects of our training materials led to the decision to pick them?
\item Why did the trainee decide to use these training modules (e.g., requirement for a course, PI told them to, self-motivated to improve their research practices, interested in learning new technology)?
\end{itemize}

\textbf{Levels of evaluation:}

\begin{itemize}
\item \textbf{Final users of the online book and videos.} These could potentially be from anywhere in the world, and for many we won't have great ways to contact them. 
\item \textbf{Workshop attendees for the workshops we plan to propose and do at national microbiology / immunology conferences.} For these people, we could definitely do a survey before to get information on demographics, education level, interest in the training materials, etc. We could also do a post survey to find out what they learned, how helpful it was, what they found confusing, etc. Finally, we could get their email addresses to ask some longer-term evaluation questions (e.g., How are they using what they learned 1--2 years after taking the workshop? How much did they retain in terms of principals, implementation, and examples?). We can also use questions that are asked during the workshops and areas where additional materials (applied exercises, quizzes, discussion questions) are problematic to help us hone our training materials.
\item \textbf{Early online users.} We will plan to develop and post the text and some of the additional educational materials (e.g., quizzes, discussion questions) online through GitHub \textit{as we write the book and develop the materials}. We will use social media to invite people to try out the book as we develop it. Based on previous work developing online books, we have found that this open development process can help attract users very early in the process, and that these users are often very helpful in providing feedback as the book is developed. We will elicit their feedback through GitHub (``Issues" page will be the main forum for them to post comments and suggestions).
\item \textbf{CSU pilot users.} We can ask these pilot users many questions, both before and after the pilot testing. Further, we will have access to ask them longer-term outcomes, as well as to ask at the department level how the use of these training materials by a number of people in the department has changed research practices and what is considered ``best practice" for research in the department (i.e., a `bubble up" effect).
\item \textbf{Pilot users from other institutions.} Similar to CSU pilot users, although we'll have a bit less access for determining longer-term and department-wide outcomes.
\end{itemize}

\subsection{Dissemination Plan}

\begin{quotation}
``A specific plan must be provided to disseminate the finished training modules \textbf{nationally} and make them \textbf{freely accessible}. In addition, links to these modules will be posted and maintained on the NIGMS web site."
\end{quotation}

We will create an online tutorial book, since providing tutorials, example code, and example datasets can substantially improve the ability of new users to learn software tools \cite{list2017ten}. We will use GitHub's free ``Pages" web publishing framework to publish the book freely online, and we will also submit it to the \textit{bookdown.org} website under the Creative Commons Attribution-ShareAlike 4.0 International License [ref]. Dr. Anderson (PI) has previously created two \textit{bookdown}-based books, \textit{R Programming for Research} and \textit{Mastering Software Development in R}. Both are publicly and freely available online under the Creative Commons license (see LOS, Xie). 

We will publish the video lectures using the YouTube platform and embed these videos within the online book. The videos, like the book, will be published under a Creative Commons license.

For many online training materials on the principals and implementation of computationally reproducible research, the target audience is statisticians and other researchers who focus on data analysis. This audience now has access to many excellent resources for improving research reproducibility at the data analysis stage. Our target audience is instead researchers who focus on conducting experimental, laboratory-based biomedical research and whose research tasks are more focused on the earlier research steps of running experiments, recording experimental data, and pre-processing that data, prior to data analysis steps. We are focusing our dissemination plans on this target audience, for whom training materials on improving the reproducibility of later data analysis steps might be of limited relevance.

We will also take steps to make sure that our target audience---laboratory-based biomedical scientists---are aware of these training materials. We will travel in Years 02 and 03 to one national microbiology ([conference]) and one national immunology ([conference]) conference. We will submit proposals to these conferences to present half-day (student?) workshops covering why and how to improve reproducibility in experimental data recording and pre-processing for biomedical microbiology and immunology research. We will also submit poster abstracts to present and discuss the training materials as part of each conference's poster sessions. We have budgeted for two members of our team (the PI and one co-I) to attend each of these conferences to help disseminate the training materials produced by the project. [CSU's microbiology seminar series? Conferences on-campus at CSU? Are there ones that are repeated every year?]

The PI has previously had substantial success in disseminating online training materials. She is the co-instructor of a five-course specialization on \textit{Mastering Software Development in R} through the Massive Open Online Course platform Coursera. The first course in this series has had over 25,000 participants since it was opened in fall 2016. An accompanying online book on the LeanPub platform has been accessed by [x] people. The PI, however, does not have a presence in the microbiology and immunology community, and so the co-investigators, who are heavily involved in this community, will form a key bridge in making sure our target audience is aware that these training materials are available.
    
\subsection{Timeline}

\begin{quotation}
``Provide a timeline for \textbf{module development}, \textbf{piloting and refinement}, \textbf{dissemination}, \textbf{evaluation}, and \textbf{maintenance}.  This timeline must propose making the training publicly available within two years of the award date."
\end{quotation}


\begin{figure}[ht]
    \includegraphics[width=\textwidth]{figures/timeline.pdf}
    \caption{Timeline for proposed activities under this funding.}
    \label{fig:timeline}
\end{figure}



\clearpage

\bibliographystyle{unsrtnat}
\bibliography{rep_modules}

\end{document}